spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -552,6 +584,80 @@ public static class TestPojoWithAnnotatedArguments {
 Dispatching via `@StreamListener` conditions is only supported for handlers of individual messages, and not for reactive programming support (described below).
 ====
 
+===== Using Polled Consumers
+
+When using polled consumers, you poll the `PollableMessageSource` on demand.
+For example, given...
+
+[source,java]
+----
+public interface PolledConsumer {
+
+    @Input
+    PollableMessageSource dest1In();
+
+    @Output
+    MessageChannel dest2Out();
+
+}
+----
+
+...you might use that consumer as follows:
+
+[source,java]
+----
+@Bean
+public ApplicationRunner poller(PollableMessageSource dest1In, MessageChannel dest2Out) {
+    return args -> {
+        while (someCondition()) {
+            try {
+                if (!dest1In.poll(m -> {
+                    String newPayload = ((String) m.getPayload()).toUpperCase();
+                    dest2Out.send(new GenericMessage<>(newPayload));
+                })) {
+                    Thread.sleep(1000);
+                }
+            }
+            catch (Exception e) {
+                // handle failure
+            }
+        }
+    };
+}
+----
+
+The `PollableMessageSource.poll()` method takes a `MessageHandler` argument (often a lambda expression as shown here).
+It returns `true` if the message was received and successfully processed.
+
+As with message-driven consumers, if the `MessageHandler` throws an exception, messages are published to error channels as discussed in <<binder-error-channels>>.
```

Comment:
Gary, just trying to correlate this sentence to the above code and thinking it may be a bit confusing. The code above is already written to handle exception. May be remove` try/catch`? (I know you need it to `Thread.sleep()`, but for this example you can probably use `LockSupport` or something)

Reply:
maybe 

    // handle failure (throw an exception to reject the message)

Reply:
I'm ok with that